### PR TITLE
Nexus: support LLNL machines Lassen and Ruby

### PR DESCRIPTION
**Pull request by jtkrogel**
## Proposed changes

Support running Nexus on current LLNL machines Lassen and Ruby.  Nexus tests pass.

## What type(s) of changes does this code introduce?

- Other (please describe): machine support

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

Lassen/Ruby/laptop

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed

### CODE REVIEW

**Review by prckent — DISMISSED** [review]
LGTM

Would be wonderful to have a guide for external folks on how to write these...

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
@jtkrogel there is a merge conflict to be resolved.

**Comment by jtkrogel** [comment]
@ye-luo @prckent conflicts should be resolved now.

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

File: nexus/tests/unit/test_machines.py [hunk 1]
```diff
@@ -1269,18 +1269,30 @@ def job_commands_equal(c1,c2):
...
         ('kestrel'        , 'n2_t2'         ) : 'srun test.x',
         ('kestrel'        , 'n2_t2_e'       ) : 'srun test.x',
         ('kestrel'        , 'n2_t2_p2'      ) : 'srun test.x',
-        ('inti'           , 'n1'            ) : 'mpirun -np 64 test.x',
-        ('inti'           , 'n1_p1'         ) : 'mpirun -np 1 test.x',
-        ('inti'           , 'n2'            ) : 'mpirun -np 128 test.x',
-        ('inti'           , 'n2_t2'         ) : 'mpirun -np 64 test.x',
-        ('inti'           , 'n2_t2_e'       ) : 'mpirun -np 64 test.x',
-        ('inti'           , 'n2_t2_p2'      ) : 'mpirun -np 4 test.x',
+        ('inti'           , 'n1'            ) : 'srun test.x',
```
Comment (prckent) [inline nexus/tests/unit/test_machines.py:1290]: Switch of mpirun -> srun was intentional for inti?

**Review by prckent — APPROVED** [review]

File: nexus/tests/unit/test_machines.py (hunk 1 above)
Comment (jtkrogel) [reply nexus/tests/unit/test_machines.py:1290]: Yes

**Pull request merged**